lib/Dialect/FIRRTL/Transforms/PrefixModules.cpp

```diff
@@ -170,14 +171,24 @@ void PrefixModulesPass::renameModuleBody(std::string prefix, FModuleOp module) {
               // Iterate over the modules of the NonLocalAnchor op, and update
               // it.
               SmallVector<Attribute, 4> newMods;
-              for (auto oldMod : nlaOp.modpath()) {
-                if (instanceOp.moduleNameAttr() ==
-                    oldMod.cast<FlatSymbolRefAttr>())
-                  newMods.push_back(FlatSymbolRefAttr::get(context, newTarget));
-                else
-                  newMods.push_back(oldMod.cast<FlatSymbolRefAttr>());
+              for (auto nameRef : nlaOp.namepath()) {
+                if (auto oldMod = nameRef.dyn_cast<hw::InnerRefAttr>()) {
+                  if (instanceOp.moduleName().equals(
+                          oldMod.getModule().getValue()))
```

Comment:
you can do a pointer compare instead of string cmp with
```c++
instanceOp.moduleName().getAttr() == oldModule.getModule()
```

Reply:
Thanks, fixed it.